data/tools/wmlunits

```diff
@@ -33,12 +33,17 @@ def copy_images():
     shutil.copy2(os.path.join(os.path.dirname(os.path.realpath(__file__)), "unit_tree", "menu.js"),
                  options.output)
 
-def shell_out(com):
-    p = subprocess.Popen(com,
-                         stdout=subprocess.PIPE,
-                         stderr=subprocess.PIPE)
-    out, err = p.communicate()
-    return out
+def shell_out(wesnoth_exe, com):
+    base_com = [wesnoth_exe, "--nobanner"] + com
+    cli_opt_matrix = [[], ["--no-log-to-file"], ["--wnoconsole", "--wnoredirect"]]
+    for cli_opts in cli_opt_matrix:
+        com = base_com + cli_opts
+        p = subprocess.run(com, capture_output=True, text=True)
+        # Wesnoth 1.16 writes to stdout, 1.17 to stderr
```

Comment:
This is a bug. I'd rather that gets fixed instead of adding workarounds. Also the output is mangled by sanitization so it's currently broken either way.

Reply:
There are already released versions that use stderr, so we might as well not pointlessly remove compat with them.

Reply:
You don't mean the sanitization at make_name, do you?

Reply:
No, I'm just talking about wesnoth's output for `--data-path` and the like.

This is now fixed in wesnoth with 60184ff.

I think it'd be more future-proof to check the exit status for errors instead of "parsing" stderr and use stdout as the return value. (I should add something to CI to make sure --data-path output is not broken again...)